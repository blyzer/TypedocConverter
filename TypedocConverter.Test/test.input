{
	"id": 0,
	"name": "",
	"kind": 0,
	"flags": {},
	"children": [
		{
			"id": 1,
			"name": "\"1\"",
			"kind": 1,
			"kindString": "External module",
			"flags": {
				"isExported": true
			},
			"originalName": "C:/Users/hez20/source/repos/TypedocConverter/TypedocConverter/bin/Release/netcoreapp3.1/win-x64/publish/1.ts",
			"children": [
				{
					"id": 2,
					"name": "test",
					"kind": 2,
					"kindString": "Module",
					"flags": {
						"isExported": true
					},
					"children": [
						{
							"id": 3,
							"name": "MyEnum",
							"kind": 4,
							"kindString": "Enumeration",
							"flags": {
								"isExported": true
							},
							"comment": {
								"shortText": "The declaration of an enum"
							},
							"children": [
								{
									"id": 4,
									"name": "A",
									"kind": 16,
									"kindString": "Enumeration member",
									"flags": {
										"isExported": true
									},
									"sources": [
										{
											"fileName": "1.ts",
											"line": 6,
											"character": 5
										}
									],
									"defaultValue": "0"
								},
								{
									"id": 5,
									"name": "B",
									"kind": 16,
									"kindString": "Enumeration member",
									"flags": {
										"isExported": true
									},
									"sources": [
										{
											"fileName": "1.ts",
											"line": 7,
											"character": 5
										}
									],
									"defaultValue": "1"
								},
								{
									"id": 6,
									"name": "C",
									"kind": 16,
									"kindString": "Enumeration member",
									"flags": {
										"isExported": true
									},
									"sources": [
										{
											"fileName": "1.ts",
											"line": 8,
											"character": 5
										}
									],
									"defaultValue": "2"
								},
								{
									"id": 7,
									"name": "D",
									"kind": 16,
									"kindString": "Enumeration member",
									"flags": {
										"isExported": true
									},
									"sources": [
										{
											"fileName": "1.ts",
											"line": 9,
											"character": 5
										}
									],
									"defaultValue": "C"
								}
							],
							"groups": [
								{
									"title": "Enumeration members",
									"kind": 16,
									"children": [
										4,
										5,
										6,
										7
									]
								}
							],
							"sources": [
								{
									"fileName": "1.ts",
									"line": 5,
									"character": 20
								}
							]
						},
						{
							"id": 37,
							"name": "MyClass1",
							"kind": 128,
							"kindString": "Class",
							"flags": {
								"isExported": true
							},
							"comment": {
								"shortText": "The declaration of a class"
							},
							"typeParameter": [
								{
									"id": 38,
									"name": "T",
									"kind": 131072,
									"kindString": "Type parameter",
									"flags": {
										"isExported": true
									}
								}
							],
							"children": [
								{
									"id": 45,
									"name": "onTest",
									"kind": 8388608,
									"kindString": "Event",
									"flags": {
										"isExported": true
									},
									"signatures": [
										{
											"id": 46,
											"name": "onTest",
											"kind": 8388608,
											"kindString": "Event",
											"flags": {
												"isExported": true
											},
											"comment": {
												"shortText": "An event"
											},
											"parameters": [
												{
													"id": 47,
													"name": "listener",
													"kind": 32768,
													"kindString": "Parameter",
													"flags": {
														"isExported": true
													},
													"type": {
														"type": "reflection",
														"declaration": {
															"id": 48,
															"name": "__type",
															"kind": 65536,
															"kindString": "Type literal",
															"flags": {
																"isExported": true
															},
															"signatures": [
																{
																	"id": 49,
																	"name": "__call",
																	"kind": 4096,
																	"kindString": "Call signature",
																	"flags": {
																		"isExported": true
																	},
																	"parameters": [
																		{
																			"id": 50,
																			"name": "e",
																			"kind": 32768,
																			"kindString": "Parameter",
																			"flags": {
																				"isExported": true
																			},
																			"type": {
																				"type": "reference",
																				"id": 8,
																				"name": "MyInterface1"
																			}
																		}
																	],
																	"type": {
																		"type": "intrinsic",
																		"name": "void"
																	}
																}
															],
															"sources": [
																{
																	"fileName": "1.ts",
																	"line": 62,
																	"character": 20
																}
															]
														}
													}
												}
											],
											"type": {
												"type": "intrinsic",
												"name": "void"
											}
										}
									],
									"sources": [
										{
											"fileName": "1.ts",
											"line": 62,
											"character": 10
										}
									],
									"implementationOf": {
										"type": "reference",
										"id": 15,
										"name": "MyInterface1.onTest"
									}
								},
								{
									"id": 51,
									"name": "testProp",
									"kind": 1024,
									"kindString": "Property",
									"flags": {
										"isExported": true
									},
									"comment": {
										"shortText": "An property"
									},
									"sources": [
										{
											"fileName": "1.ts",
											"line": 66,
											"character": 21
										}
									],
									"type": {
										"type": "intrinsic",
										"name": "string"
									},
									"implementationOf": {
										"type": "reference",
										"id": 21,
										"name": "MyInterface1.testProp"
									}
								},
								{
									"id": 39,
									"name": "testMethod",
									"kind": 2048,
									"kindString": "Method",
									"flags": {
										"isExported": true
									},
									"signatures": [
										{
											"id": 40,
											"name": "testMethod",
											"kind": 4096,
											"kindString": "Call signature",
											"flags": {
												"isExported": true
											},
											"comment": {
												"shortText": "A method"
											},
											"parameters": [
												{
													"id": 41,
													"name": "arg",
													"kind": 32768,
													"kindString": "Parameter",
													"flags": {
														"isExported": true
													},
													"type": {
														"type": "intrinsic",
														"name": "string"
													}
												},
												{
													"id": 42,
													"name": "callback",
													"kind": 32768,
													"kindString": "Parameter",
													"flags": {
														"isExported": true
													},
													"type": {
														"type": "reflection",
														"declaration": {
															"id": 43,
															"name": "__type",
															"kind": 65536,
															"kindString": "Type literal",
															"flags": {
																"isExported": true
															},
															"signatures": [
																{
																	"id": 44,
																	"name": "__call",
																	"kind": 4096,
																	"kindString": "Call signature",
																	"flags": {
																		"isExported": true
																	},
																	"type": {
																		"type": "intrinsic",
																		"name": "void"
																	}
																}
															],
															"sources": [
																{
																	"fileName": "1.ts",
																	"line": 57,
																	"character": 37
																}
															]
														}
													}
												}
											],
											"type": {
												"type": "intrinsic",
												"name": "string"
											}
										}
									],
									"sources": [
										{
											"fileName": "1.ts",
											"line": 57,
											"character": 14
										}
									],
									"implementationOf": {
										"type": "reference",
										"id": 9,
										"name": "MyInterface1.testMethod"
									}
								},
								{
									"id": 52,
									"name": "staticMethod",
									"kind": 2048,
									"kindString": "Method",
									"flags": {
										"isStatic": true,
										"isExported": true
									},
									"signatures": [
										{
											"id": 53,
											"name": "staticMethod",
											"kind": 4096,
											"kindString": "Call signature",
											"flags": {
												"isExported": true
											},
											"parameters": [
												{
													"id": 54,
													"name": "value",
													"kind": 32768,
													"kindString": "Parameter",
													"flags": {
														"isExported": true
													},
													"type": {
														"type": "intrinsic",
														"name": "string"
													}
												},
												{
													"id": 55,
													"name": "isOption",
													"kind": 32768,
													"kindString": "Parameter",
													"flags": {
														"isExported": true,
														"isOptional": true
													},
													"type": {
														"type": "intrinsic",
														"name": "boolean"
													}
												}
											],
											"type": {
												"type": "reference",
												"id": 75,
												"name": "UnionStr"
											}
										}
									],
									"sources": [
										{
											"fileName": "1.ts",
											"line": 67,
											"character": 23
										}
									]
								}
							],
							"groups": [
								{
									"title": "Events",
									"kind": 8388608,
									"children": [
										45
									]
								},
								{
									"title": "Properties",
									"kind": 1024,
									"children": [
										51
									]
								},
								{
									"title": "Methods",
									"kind": 2048,
									"children": [
										39,
										52
									]
								}
							],
							"sources": [
								{
									"fileName": "1.ts",
									"line": 53,
									"character": 23
								}
							],
							"implementedTypes": [
								{
									"type": "reference",
									"id": 8,
									"name": "MyInterface1"
								}
							]
						},
						{
							"id": 56,
							"name": "MyClass2",
							"kind": 128,
							"kindString": "Class",
							"flags": {
								"isExported": true
							},
							"comment": {
								"shortText": "Another declaration of a class"
							},
							"typeParameter": [
								{
									"id": 57,
									"name": "T",
									"kind": 131072,
									"kindString": "Type parameter",
									"flags": {
										"isExported": true
									}
								}
							],
							"children": [
								{
									"id": 64,
									"name": "onTest",
									"kind": 8388608,
									"kindString": "Event",
									"flags": {
										"isExported": true
									},
									"signatures": [
										{
											"id": 65,
											"name": "onTest",
											"kind": 8388608,
											"kindString": "Event",
											"flags": {
												"isExported": true
											},
											"comment": {
												"shortText": "An event"
											},
											"parameters": [
												{
													"id": 66,
													"name": "listener",
													"kind": 32768,
													"kindString": "Parameter",
													"flags": {
														"isExported": true
													},
													"type": {
														"type": "reflection",
														"declaration": {
															"id": 67,
															"name": "__type",
															"kind": 65536,
															"kindString": "Type literal",
															"flags": {
																"isExported": true
															},
															"signatures": [
																{
																	"id": 68,
																	"name": "__call",
																	"kind": 4096,
																	"kindString": "Call signature",
																	"flags": {
																		"isExported": true
																	},
																	"parameters": [
																		{
																			"id": 69,
																			"name": "e",
																			"kind": 32768,
																			"kindString": "Parameter",
																			"flags": {
																				"isExported": true
																			},
																			"type": {
																				"type": "reference",
																				"id": 22,
																				"typeArguments": [
																					{
																						"type": "typeParameter",
																						"name": "T"
																					}
																				],
																				"name": "MyInterface2"
																			}
																		}
																	],
																	"type": {
																		"type": "intrinsic",
																		"name": "void"
																	}
																}
															],
															"sources": [
																{
																	"fileName": "1.ts",
																	"line": 82,
																	"character": 20
																}
															]
														}
													}
												}
											],
											"type": {
												"type": "intrinsic",
												"name": "void"
											}
										}
									],
									"sources": [
										{
											"fileName": "1.ts",
											"line": 82,
											"character": 10
										}
									],
									"implementationOf": {
										"type": "reference",
										"id": 30,
										"name": "MyInterface2.onTest"
									}
								},
								{
									"id": 70,
									"name": "testProp",
									"kind": 1024,
									"kindString": "Property",
									"flags": {
										"isExported": true
									},
									"comment": {
										"shortText": "An property"
									},
									"sources": [
										{
											"fileName": "1.ts",
											"line": 86,
											"character": 21
										}
									],
									"type": {
										"type": "typeParameter",
										"name": "T"
									},
									"implementationOf": {
										"type": "reference",
										"id": 36,
										"name": "MyInterface2.testProp"
									}
								},
								{
									"id": 58,
									"name": "testMethod",
									"kind": 2048,
									"kindString": "Method",
									"flags": {
										"isExported": true
									},
									"signatures": [
										{
											"id": 59,
											"name": "testMethod",
											"kind": 4096,
											"kindString": "Call signature",
											"flags": {
												"isExported": true
											},
											"comment": {
												"shortText": "A method"
											},
											"parameters": [
												{
													"id": 60,
													"name": "arg",
													"kind": 32768,
													"kindString": "Parameter",
													"flags": {
														"isExported": true
													},
													"type": {
														"type": "typeParameter",
														"name": "T"
													}
												},
												{
													"id": 61,
													"name": "callback",
													"kind": 32768,
													"kindString": "Parameter",
													"flags": {
														"isExported": true
													},
													"type": {
														"type": "reflection",
														"declaration": {
															"id": 62,
															"name": "__type",
															"kind": 65536,
															"kindString": "Type literal",
															"flags": {
																"isExported": true
															},
															"signatures": [
																{
																	"id": 63,
																	"name": "__call",
																	"kind": 4096,
																	"kindString": "Call signature",
																	"flags": {
																		"isExported": true
																	},
																	"type": {
																		"type": "intrinsic",
																		"name": "void"
																	}
																}
															],
															"sources": [
																{
																	"fileName": "1.ts",
																	"line": 77,
																	"character": 32
																}
															]
														}
													}
												}
											],
											"type": {
												"type": "typeParameter",
												"name": "T"
											}
										}
									],
									"sources": [
										{
											"fileName": "1.ts",
											"line": 77,
											"character": 14
										}
									],
									"implementationOf": {
										"type": "reference",
										"id": 24,
										"name": "MyInterface2.testMethod"
									}
								},
								{
									"id": 71,
									"name": "staticMethod",
									"kind": 2048,
									"kindString": "Method",
									"flags": {
										"isStatic": true,
										"isExported": true
									},
									"signatures": [
										{
											"id": 72,
											"name": "staticMethod",
											"kind": 4096,
											"kindString": "Call signature",
											"flags": {
												"isExported": true
											},
											"parameters": [
												{
													"id": 73,
													"name": "value",
													"kind": 32768,
													"kindString": "Parameter",
													"flags": {
														"isExported": true
													},
													"type": {
														"type": "intrinsic",
														"name": "string"
													}
												},
												{
													"id": 74,
													"name": "isOption",
													"kind": 32768,
													"kindString": "Parameter",
													"flags": {
														"isExported": true,
														"isOptional": true
													},
													"type": {
														"type": "intrinsic",
														"name": "boolean"
													}
												}
											],
											"type": {
												"type": "reference",
												"id": 75,
												"name": "UnionStr"
											}
										}
									],
									"sources": [
										{
											"fileName": "1.ts",
											"line": 87,
											"character": 23
										}
									]
								}
							],
							"groups": [
								{
									"title": "Events",
									"kind": 8388608,
									"children": [
										64
									]
								},
								{
									"title": "Properties",
									"kind": 1024,
									"children": [
										70
									]
								},
								{
									"title": "Methods",
									"kind": 2048,
									"children": [
										58,
										71
									]
								}
							],
							"sources": [
								{
									"fileName": "1.ts",
									"line": 73,
									"character": 23
								}
							],
							"implementedTypes": [
								{
									"type": "reference",
									"id": 22,
									"typeArguments": [
										{
											"type": "typeParameter",
											"name": "T"
										}
									],
									"name": "MyInterface2"
								}
							]
						},
						{
							"id": 8,
							"name": "MyInterface1",
							"kind": 256,
							"kindString": "Interface",
							"flags": {
								"isExported": true
							},
							"comment": {
								"shortText": "The declaration of an interface"
							},
							"children": [
								{
									"id": 15,
									"name": "onTest",
									"kind": 8388608,
									"kindString": "Event",
									"flags": {
										"isExported": true
									},
									"signatures": [
										{
											"id": 16,
											"name": "onTest",
											"kind": 8388608,
											"kindString": "Event",
											"flags": {
												"isExported": true
											},
											"comment": {
												"shortText": "An event"
											},
											"parameters": [
												{
													"id": 17,
													"name": "listener",
													"kind": 32768,
													"kindString": "Parameter",
													"flags": {
														"isExported": true
													},
													"type": {
														"type": "reflection",
														"declaration": {
															"id": 18,
															"name": "__type",
															"kind": 65536,
															"kindString": "Type literal",
															"flags": {
																"isExported": true
															},
															"signatures": [
																{
																	"id": 19,
																	"name": "__call",
																	"kind": 4096,
																	"kindString": "Call signature",
																	"flags": {
																		"isExported": true
																	},
																	"parameters": [
																		{
																			"id": 20,
																			"name": "e",
																			"kind": 32768,
																			"kindString": "Parameter",
																			"flags": {
																				"isExported": true
																			},
																			"type": {
																				"type": "reference",
																				"id": 8,
																				"name": "MyInterface1"
																			}
																		}
																	],
																	"type": {
																		"type": "intrinsic",
																		"name": "void"
																	}
																}
															],
															"sources": [
																{
																	"fileName": "1.ts",
																	"line": 24,
																	"character": 20
																}
															]
														}
													}
												}
											],
											"type": {
												"type": "intrinsic",
												"name": "void"
											}
										}
									],
									"sources": [
										{
											"fileName": "1.ts",
											"line": 24,
											"character": 10
										}
									]
								},
								{
									"id": 21,
									"name": "testProp",
									"kind": 1024,
									"kindString": "Property",
									"flags": {
										"isExported": true
									},
									"comment": {
										"shortText": "An property"
									},
									"sources": [
										{
											"fileName": "1.ts",
											"line": 28,
											"character": 21
										}
									],
									"type": {
										"type": "intrinsic",
										"name": "string"
									}
								},
								{
									"id": 9,
									"name": "testMethod",
									"kind": 2048,
									"kindString": "Method",
									"flags": {
										"isExported": true
									},
									"signatures": [
										{
											"id": 10,
											"name": "testMethod",
											"kind": 4096,
											"kindString": "Call signature",
											"flags": {
												"isExported": true
											},
											"comment": {
												"shortText": "A method"
											},
											"parameters": [
												{
													"id": 11,
													"name": "arg",
													"kind": 32768,
													"kindString": "Parameter",
													"flags": {
														"isExported": true
													},
													"type": {
														"type": "intrinsic",
														"name": "string"
													}
												},
												{
													"id": 12,
													"name": "callback",
													"kind": 32768,
													"kindString": "Parameter",
													"flags": {
														"isExported": true
													},
													"type": {
														"type": "reflection",
														"declaration": {
															"id": 13,
															"name": "__type",
															"kind": 65536,
															"kindString": "Type literal",
															"flags": {
																"isExported": true
															},
															"signatures": [
																{
																	"id": 14,
																	"name": "__call",
																	"kind": 4096,
																	"kindString": "Call signature",
																	"flags": {
																		"isExported": true
																	},
																	"type": {
																		"type": "intrinsic",
																		"name": "void"
																	}
																}
															],
															"sources": [
																{
																	"fileName": "1.ts",
																	"line": 19,
																	"character": 37
																}
															]
														}
													}
												}
											],
											"type": {
												"type": "intrinsic",
												"name": "string"
											}
										}
									],
									"sources": [
										{
											"fileName": "1.ts",
											"line": 19,
											"character": 14
										}
									]
								}
							],
							"groups": [
								{
									"title": "Events",
									"kind": 8388608,
									"children": [
										15
									]
								},
								{
									"title": "Properties",
									"kind": 1024,
									"children": [
										21
									]
								},
								{
									"title": "Methods",
									"kind": 2048,
									"children": [
										9
									]
								}
							],
							"sources": [
								{
									"fileName": "1.ts",
									"line": 15,
									"character": 31
								}
							],
							"implementedBy": [
								{
									"type": "reference",
									"id": 37,
									"name": "MyClass1"
								}
							]
						},
						{
							"id": 22,
							"name": "MyInterface2",
							"kind": 256,
							"kindString": "Interface",
							"flags": {
								"isExported": true
							},
							"comment": {
								"shortText": "Another declaration of an interface"
							},
							"typeParameter": [
								{
									"id": 23,
									"name": "T",
									"kind": 131072,
									"kindString": "Type parameter",
									"flags": {
										"isExported": true
									}
								}
							],
							"children": [
								{
									"id": 30,
									"name": "onTest",
									"kind": 8388608,
									"kindString": "Event",
									"flags": {
										"isExported": true
									},
									"signatures": [
										{
											"id": 31,
											"name": "onTest",
											"kind": 8388608,
											"kindString": "Event",
											"flags": {
												"isExported": true
											},
											"comment": {
												"shortText": "An event"
											},
											"parameters": [
												{
													"id": 32,
													"name": "listener",
													"kind": 32768,
													"kindString": "Parameter",
													"flags": {
														"isExported": true
													},
													"type": {
														"type": "reflection",
														"declaration": {
															"id": 33,
															"name": "__type",
															"kind": 65536,
															"kindString": "Type literal",
															"flags": {
																"isExported": true
															},
															"signatures": [
																{
																	"id": 34,
																	"name": "__call",
																	"kind": 4096,
																	"kindString": "Call signature",
																	"flags": {
																		"isExported": true
																	},
																	"parameters": [
																		{
																			"id": 35,
																			"name": "e",
																			"kind": 32768,
																			"kindString": "Parameter",
																			"flags": {
																				"isExported": true
																			},
																			"type": {
																				"type": "reference",
																				"id": 22,
																				"typeArguments": [
																					{
																						"type": "typeParameter",
																						"name": "T"
																					}
																				],
																				"name": "MyInterface2"
																			}
																		}
																	],
																	"type": {
																		"type": "intrinsic",
																		"name": "void"
																	}
																}
															],
															"sources": [
																{
																	"fileName": "1.ts",
																	"line": 43,
																	"character": 20
																}
															]
														}
													}
												}
											],
											"type": {
												"type": "intrinsic",
												"name": "void"
											}
										}
									],
									"sources": [
										{
											"fileName": "1.ts",
											"line": 43,
											"character": 10
										}
									]
								},
								{
									"id": 36,
									"name": "testProp",
									"kind": 1024,
									"kindString": "Property",
									"flags": {
										"isExported": true
									},
									"comment": {
										"shortText": "An property"
									},
									"sources": [
										{
											"fileName": "1.ts",
											"line": 47,
											"character": 21
										}
									],
									"type": {
										"type": "typeParameter",
										"name": "T"
									}
								},
								{
									"id": 24,
									"name": "testMethod",
									"kind": 2048,
									"kindString": "Method",
									"flags": {
										"isExported": true
									},
									"signatures": [
										{
											"id": 25,
											"name": "testMethod",
											"kind": 4096,
											"kindString": "Call signature",
											"flags": {
												"isExported": true
											},
											"comment": {
												"shortText": "A method"
											},
											"parameters": [
												{
													"id": 26,
													"name": "arg",
													"kind": 32768,
													"kindString": "Parameter",
													"flags": {
														"isExported": true
													},
													"type": {
														"type": "typeParameter",
														"name": "T"
													}
												},
												{
													"id": 27,
													"name": "callback",
													"kind": 32768,
													"kindString": "Parameter",
													"flags": {
														"isExported": true
													},
													"type": {
														"type": "reflection",
														"declaration": {
															"id": 28,
															"name": "__type",
															"kind": 65536,
															"kindString": "Type literal",
															"flags": {
																"isExported": true
															},
															"signatures": [
																{
																	"id": 29,
																	"name": "__call",
																	"kind": 4096,
																	"kindString": "Call signature",
																	"flags": {
																		"isExported": true
																	},
																	"type": {
																		"type": "intrinsic",
																		"name": "void"
																	}
																}
															],
															"sources": [
																{
																	"fileName": "1.ts",
																	"line": 38,
																	"character": 32
																}
															]
														}
													}
												}
											],
											"type": {
												"type": "typeParameter",
												"name": "T"
											}
										}
									],
									"sources": [
										{
											"fileName": "1.ts",
											"line": 38,
											"character": 14
										}
									]
								}
							],
							"groups": [
								{
									"title": "Events",
									"kind": 8388608,
									"children": [
										30
									]
								},
								{
									"title": "Properties",
									"kind": 1024,
									"children": [
										36
									]
								},
								{
									"title": "Methods",
									"kind": 2048,
									"children": [
										24
									]
								}
							],
							"sources": [
								{
									"fileName": "1.ts",
									"line": 34,
									"character": 31
								}
							],
							"implementedBy": [
								{
									"type": "reference",
									"id": 56,
									"name": "MyClass2"
								}
							]
						},
						{
							"id": 75,
							"name": "UnionStr",
							"kind": 4194304,
							"kindString": "Type alias",
							"flags": {
								"isExported": true
							},
							"comment": {
								"shortText": "The declaration of a type alias"
							},
							"sources": [
								{
									"fileName": "1.ts",
									"line": 93,
									"character": 22
								}
							],
							"type": {
								"type": "union",
								"types": [
									{
										"type": "stringLiteral",
										"value": "A"
									},
									{
										"type": "stringLiteral",
										"value": "B"
									},
									{
										"type": "stringLiteral",
										"value": "C"
									},
									{
										"type": "stringLiteral",
										"value": "other"
									}
								]
							}
						}
					],
					"groups": [
						{
							"title": "Enumerations",
							"kind": 4,
							"children": [
								3
							]
						},
						{
							"title": "Classes",
							"kind": 128,
							"children": [
								37,
								56
							]
						},
						{
							"title": "Interfaces",
							"kind": 256,
							"children": [
								8,
								22
							]
						},
						{
							"title": "Type aliases",
							"kind": 4194304,
							"children": [
								75
							]
						}
					],
					"sources": [
						{
							"fileName": "1.ts",
							"line": 1,
							"character": 22
						}
					]
				}
			],
			"groups": [
				{
					"title": "Modules",
					"kind": 2,
					"children": [
						2
					]
				}
			],
			"sources": [
				{
					"fileName": "1.ts",
					"line": 1,
					"character": 0
				}
			]
		}
	],
	"groups": [
		{
			"title": "External modules",
			"kind": 1,
			"children": [
				1
			]
		}
	]
}