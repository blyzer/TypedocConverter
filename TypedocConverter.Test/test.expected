namespace TypedocConverter.Test
{

    /// <summary>
    /// The declaration of an enum
    /// </summary>
    enum MyEnum
    {
        [Newtonsoft.Json.JsonProperty("A", NullValueHandling = Newtonsoft.Json.NullValueHandling.Ignore)]
        A = 0,
        [Newtonsoft.Json.JsonProperty("B", NullValueHandling = Newtonsoft.Json.NullValueHandling.Ignore)]
        B = 1,
        [Newtonsoft.Json.JsonProperty("C", NullValueHandling = Newtonsoft.Json.NullValueHandling.Ignore)]
        C = 2,
        [Newtonsoft.Json.JsonProperty("D", NullValueHandling = Newtonsoft.Json.NullValueHandling.Ignore)]
        D = 2
    }
}

namespace TypedocConverter.Test
{

    /// <summary>
    /// The declaration of a class
    /// </summary>
    class MyClass1<T> : MyInterface1
    {
        /// <summary>
        /// An property
        /// </summary>
        [Newtonsoft.Json.JsonProperty("testProp", NullValueHandling = Newtonsoft.Json.NullValueHandling.Ignore)]
        string TestProp { get => throw new System.NotImplementedException(); set => throw new System.NotImplementedException(); }
        /// <summary>
        /// An event
        /// </summary>
        event System.Action<MyInterface1> OnTest;
        /// <summary>
        /// A method
        /// </summary>
        string TestMethod(string arg, System.Action callback) => throw new System.NotImplementedException();
        static UnionStr StaticMethod(string value, bool isOption) => throw new System.NotImplementedException();
    }
}

namespace TypedocConverter.Test
{

    /// <summary>
    /// Another declaration of a class
    /// </summary>
    class MyClass2<T> : MyInterface2<T>
    {
        /// <summary>
        /// An property
        /// </summary>
        [Newtonsoft.Json.JsonProperty("testProp", NullValueHandling = Newtonsoft.Json.NullValueHandling.Ignore)]
        T TestProp { get => throw new System.NotImplementedException(); set => throw new System.NotImplementedException(); }
        /// <summary>
        /// An event
        /// </summary>
        event System.Action<MyInterface2<T>> OnTest;
        /// <summary>
        /// A method
        /// </summary>
        T TestMethod(T arg, System.Action callback) => throw new System.NotImplementedException();
        static UnionStr StaticMethod(string value, bool isOption) => throw new System.NotImplementedException();
    }
}

namespace TypedocConverter.Test
{

    /// <summary>
    /// The declaration of an interface
    /// </summary>
    interface MyInterface1
    {
        /// <summary>
        /// An property
        /// </summary>
        [Newtonsoft.Json.JsonProperty("testProp", NullValueHandling = Newtonsoft.Json.NullValueHandling.Ignore)]
        string TestProp { get; set; }
        /// <summary>
        /// An event
        /// </summary>
        event System.Action<MyInterface1> OnTest;
        /// <summary>
        /// A method
        /// </summary>
        string TestMethod(string arg, System.Action callback);
    }
}

namespace TypedocConverter.Test
{

    /// <summary>
    /// Another declaration of an interface
    /// </summary>
    interface MyInterface2<T>
    {
        /// <summary>
        /// An property
        /// </summary>
        [Newtonsoft.Json.JsonProperty("testProp", NullValueHandling = Newtonsoft.Json.NullValueHandling.Ignore)]
        T TestProp { get; set; }
        /// <summary>
        /// An event
        /// </summary>
        event System.Action<MyInterface2<T>> OnTest;
        /// <summary>
        /// A method
        /// </summary>
        T TestMethod(T arg, System.Action callback);
    }
}

namespace TypedocConverter.Test
{

    /// <summary>
    /// The declaration of a type alias
    /// </summary>
    [Newtonsoft.Json.JsonConverter(typeof(UnionStrConverter))]
    enum UnionStr
    {
        ///<summary>
        /// A
        ///</summary>
        [Newtonsoft.Json.JsonProperty("A", NullValueHandling = Newtonsoft.Json.NullValueHandling.Ignore)]
        A,
        ///<summary>
        /// B
        ///</summary>
        [Newtonsoft.Json.JsonProperty("B", NullValueHandling = Newtonsoft.Json.NullValueHandling.Ignore)]
        B,
        ///<summary>
        /// C
        ///</summary>
        [Newtonsoft.Json.JsonProperty("C", NullValueHandling = Newtonsoft.Json.NullValueHandling.Ignore)]
        C,
        ///<summary>
        /// other
        ///</summary>
        [Newtonsoft.Json.JsonProperty("Other", NullValueHandling = Newtonsoft.Json.NullValueHandling.Ignore)]
        Other
    }
    class UnionStrConverter : Newtonsoft.Json.JsonConverter
    {
        public override bool CanConvert(System.Type t) => t == typeof(UnionStr) || t == typeof(UnionStr?);
        public override object ReadJson(Newtonsoft.Json.JsonReader reader, System.Type t, object? existingValue, Newtonsoft.Json.JsonSerializer serializer)
            => reader.TokenType switch
            {
                Newtonsoft.Json.JsonToken.String =>
                    serializer.Deserialize<string>(reader) switch
                    {
                        "A" => UnionStr.A,
                        "B" => UnionStr.B,
                        "C" => UnionStr.C,
                        "Other" => UnionStr.Other,
                        _ => throw new System.Exception("Cannot unmarshal type UnionStr")
                    },
                _ => throw new System.Exception("Cannot unmarshal type UnionStr")
            };
        public override void WriteJson(Newtonsoft.Json.JsonWriter writer, object? untypedValue, Newtonsoft.Json.JsonSerializer serializer)
        {
            if (untypedValue is null) { serializer.Serialize(writer, null); return; }
            var value = (UnionStr)untypedValue;
            switch (value)
            {
                case UnionStr.A: serializer.Serialize(writer, "A"); return;
                case UnionStr.B: serializer.Serialize(writer, "B"); return;
                case UnionStr.C: serializer.Serialize(writer, "C"); return;
                case UnionStr.Other: serializer.Serialize(writer, "Other"); return;
                default: break;
            }
            throw new System.Exception("Cannot marshal type UnionStr");
        }
    }
}

